aten/src/ATen/native/cuda/IndexKernel.cu

```diff
@@ -369,36 +365,36 @@ void take_kernel(
 
 namespace {
 
+__global__ void masked_scatter_size_check(int64_t *totalElements, int64_t srcSize) {
+  CUDA_KERNEL_ASSERT(*totalElements <= srcSize);
+}
+
 template <typename mask_t>
 void masked_scatter_cuda_impl(Tensor& self, const Tensor& mask, const Tensor& source){
   auto srcSize = source.numel();
 
-  // Determine our output size
-  auto totalElements = mask.sum().item<int64_t>();
-
-  // The number of `1` elements present in the mask must be <= the
-  // number of elements available in `src`
-  TORCH_CHECK(totalElements <= srcSize, "source nElements must be == mask `1` elements");
+  if (srcSize == 0) {
```

Comment:
Here you are returning without an error when possibly source was invalid (contained too few elements)

Reply:
I think if `srcSize == 0`, then `mask` after broadcasting should also have 0 elements as well. But if `srcSize` is not 0, then `mask.numel()` can not be 0 as well, because it can not be broadcastable.

Reply:
It should, but if users sent empty src (erroneously), then instead of getting en error (like today), they'd get not updated self. Here's today's behavior, and there won't be an error after this change you are making: 
```
In [2]: a=torch.randn(4,4)

In [3]: mask = a>0

In [4]: vals = torch.randn(0)

In [5]: vals.size()
Out[5]: torch.Size([0])

In [6]: torch.masked_scatter(a, mask, vals)
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
<ipython-input-6-35daff38b05a> in <module>
----> 1 torch.masked_scatter(a, mask, vals)

RuntimeError: Number of elements of source < number of ones in mask
```

Reply:
Oh, I see, I thought I was checking `self.numel() == 0`. Thanks for figuring out.

Reply:
fixed now. I changed it to `self.numel() == 0`